edgedb.go

```diff
@@ -33,8 +34,53 @@ import (
 	"github.com/edgedb/edgedb-go/internal/soc"
 )
 
+const defaultMaxTxRetries = 3
+
 var rnd = rand.New(rand.NewSource(time.Now().UnixNano()))
 
+func defaultBackoff(attempt int) time.Duration {
+	backoff := math.Pow(2.0, float64(attempt)) * 100.0
+	jitter := rnd.Float64() * 100.0
+	return time.Duration(backoff+jitter) * time.Millisecond
+}
+
+// Action is work to be done in a transaction.
+type Action func(context.Context, Tx) error
+
+// Trier allows trying actions in a transaction.
+type Trier interface {
+	// TryTx runs an action in a transaction.
+	// If the action returns an error the transaction is rolled back,
+	// otherwise it is committed.
+	TryTx(context.Context, Action) error
```

Comment:
I've a question about the name. Why `TryTx` and not `TryTransaction` like in JS/Python drivers? Is `Tx` common in Go DB drivers?

Reply:
Transaction is shortened to Tx because both database/sql and pgx use this name and it seems to fit the Go naming paradigm of including the maximum information with the fewest characters.